dashboard/app/helpers/levels_helper.rb

```diff
@@ -196,6 +196,9 @@ def app_options
       @app_options = view_options.camelize_keys
     end
 
+    # Blockly caches level properties, whereas this field depends on the user
+    app_options['teacherMarkdown'] = @level.properties['teacher_markdown'] if current_user.try(:authorized_teacher?)
```

Comment:
I think this makes sense for now, but I don't know what the eventual implications are for cacheability, etc.  This isn't the first time the user is introduced while generating app_options, so it's probably not a big deal, but it seems like we want 100% cacheable levels in an ideal world, and stuff specific to authorized teachers would be fetched in a separate async request or something.  Someone more Railsy may have more to say about that (@ashercodeorg? @wjordan?)

A couple irregularities in this line of code.  You use `app_options` where surrounding code uses `@app_options`; intentional?  Likewise, you're calling `current_user` but on line 177 this method uses `@user` (maybe slightly better here).